doc: nrf5340audio: add docs for new apps (#37)
alexsven/sdk-nrf: applications/nrf5340_audio/unicast_server/README.rst

Receiving unidirectional or transceiving bidirectional audio happens using CIG and CIS.
In addition, Coordinated Set Identification Service (CSIS) is implemented on the server side.

The following limitations apply to this application:

* One CIG, one of the two CIS streams (selectable).
* Audio output: I2S/Analog headset output.
* Audio input: PDM microphone over I2S.
* Configuration: 16 bit, several bit rates ranging from 32 kbps to 124 kbps.

.. _nrf53_audio_unicast_server_app_requirements:

Requirements
************

The application shares the :ref:`requirements common to all nRF5340 Audio application <nrf53_audio_app_requirements>`.

.. _nrf53_audio_unicast_server_app_ui:

User interface
**************

Most of the user interface mappings are common across all nRF5340 Audio applications.
See the :ref:`nrf53_audio_app_ui` page for detailed overview.

This application uses specific mapping for the following user interface elements:

* Pressed on the unicast server device during playback:

  * **PLAY/PAUSE** - Starts or pauses the playback of the stream.
  * **VOL-** - Turns the playback volume down (and unmutes).
  * **VOL+** - Turns the playback volume up (and unmutes).

> * **LED1** - Blinking blue on the unicast server device: Kits have started streaming audio (BIS and CIS modes).

Reviewer: ```suggestion
* **LED1** - Blinking blue: Kits have started streaming audio.
```

* **RGB1**:

  * Solid blue - The device is programmed as the left headset.
  * Solid magenta - The device is programmed as the right headset.

.. _nrf53_audio_unicast_server_app_configuration:

Configuration
*************

See :ref:`nrf53_audio_app_configuration` and :ref:`nrf53_audio_app_fota` for configuration options common to all nRF5340 Audio applications.

For information about how to configure applications in the |NCS|, see :ref:`modifying`.

.. _nrf53_audio_unicast_server_app_building:

Building and running
********************

This application can be found under :file:`applications/nrf5340_audio/unicast_server` in the nRF Connect SDK folder structure.

The nRF5340 Audio DK comes preprogrammed with basic firmware that indicates if the kit is functional.
See :ref:`nrf53_audio_app_dk_testing_out_of_the_box` for more information.

To build the application, see :ref:`nrf53_audio_app_building`.

.. _nrf53_audio_unicast_server_app_testing:

Testing
*******

> After building and programming the application, you can test the default CIS headset mode using one :ref:`CIS gateway <nrf53_audio_unicast_client_app>` and two unicast server devices.

Reviewer: ```suggestion
After building and programming the application, you can test the default CIS headset mode using one :ref:`unicast client application <nrf53_audio_unicast_client_app>` and one or two :ref:`unicast server application <nrf53_audio_unicast_server_app>` devices.
```

> The recommended approach is to use two other nRF5340 Audio DKs programmed with the :ref:`unicast client application <nrf53_audio_unicast_client_app>` for the CIS gateway and the unicast server application for the CIS headset, respectively, but you can also use an external CIS headset device.

Reviewer: ```suggestion
The recommended approach is to use two other nRF5340 Audio DKs programmed with the :ref:`unicast client application <nrf53_audio_unicast_client_app>` for the CIS gateway and the :ref:`unicast server application <nrf53_audio_unicast_server_app>` for the CIS headset, respectively, but you can also use an external device supporting the role of unicast_server.
```

Reviewer: ah, yeah no need to reference itself :p

   After startup, this headset will be configured as the left channel headset.
   You can also just press the **RESET** button to restore the original programmed settings.

For other testing options, refer to :ref:`nrf53_audio_unicast_server_app_ui`.

After the kits have paired for the first time, they are now bonded.
This means the Long-Term Key (LTK) is stored on each side, and that the kits will only connect to each other unless the bonding information is cleared.
To clear the bonding information, press and hold **BTN 5** during boot or reprogram all the development kits.

When you finish testing, power off the nRF5340 Audio development kits by switching the power switch from On to Off.

.. _nrf53_audio_unicast_server_app_testing_steps_cis_walkie_talkie:

Testing the walkie-talkie demo
==============================

Testing the walkie-talkie demo is identical to the default testing procedure, except for the following differences:

* You must enable the Kconfig option mentioned in :ref:`nrf53_audio_app_configuration_enable_walkie_talkie` before building the application.
* Instead of controlling the playback, you can speak through the PDM microphones.

>   The line is open all the time, no need to press any buttons to talk, but the volume control works as in the default testing procedure.

Reviewer: This section is just copy-paste from unicast_client, right? Any way to link it in? That way we don't have to maintain two copies of the same text

Author: I'll do this in the next iteration.

.. _nrf53_audio_unicast_server_app:

nRF5340 Audio: Unicast server
#############################

.. contents::
   :local:
   :depth: 2

The nRF5340 Audio unicast server application implements the :ref:`CIS headset mode <nrf53_audio_app_overview>`.

In this mode, one Connected Isochronous Group (CIG) can be used with two Connected Isochronous Streams (CIS).
Receiving unidirectional or transceiving bidirectional audio happens using CIG and CIS.
In addition, Coordinated Set Identification Service (CSIS) is implemented on the server side.

The following limitations apply to this application:

* One CIG, one of the two CIS streams (selectable).
* Audio output: I2S/Analog headset output.
* Audio input: PDM microphone over I2S.
* Configuration: 16 bit, several bit rates ranging from 32 kbps to 124 kbps.

.. _nrf53_audio_unicast_server_app_requirements:

Requirements
************

The application shares the :ref:`requirements common to all nRF5340 Audio application <nrf53_audio_app_requirements>`.

.. _nrf53_audio_unicast_server_app_ui:

User interface
**************

This application uses specific mapping for the following user interface elements:

> * **VOL-** - Pressed on the headset or the CIS gateway during playback: Turns the playback volume down (and unmutes).

Reviewer: ```suggestion
* **VOL-** - Pressed on the unicast server during playback: Turns the playback volume down (and unmutes).
```
Same for the one below

Reviewer: Also add play/pause

Author: Do we need this? Isn't PLAY/PAUSE identical to all apps?

* You must enable the Kconfig option mentioned in :ref:`nrf53_audio_app_configuration_enable_walkie_talkie` before building the application.
* Instead of controlling the playback, you can speak through the PDM microphones.
  The line is open all the time, no need to press any buttons to talk, but the volume control works as in the default testing procedure.

Dependencies
************

.. note::
   The following lists mention the most important dependencies.
   For the full list, check the application's Kconfig options.
   All dependencies are automatically included.

The application uses the following |NCS| components:

* :ref:`lib_bt_ll_acs_nrf53_readme`
* :ref:`lib_contin_array`
* :ref:`lib_pcm_mix`
* :ref:`lib_tone`

This application uses the following `nrfx`_ libraries:

* :file:`nrfx_clock.h`
* :file:`nrfx_gpiote.h`
* :file:`nrfx_timer.h`
* :file:`nrfx_dppi.h`
* :file:`nrfx_i2s.h`
* :file:`nrfx_ipc.h`
* :file:`nrfx_nvmc.h`

The application also depends on the following Zephyr libraries:

* :ref:`zephyr:logging_api`
* :ref:`zephyr:kernel_api`
* :ref:`zephyr:zbus`
* :ref:`zephyr:api_peripherals`:

>    * :ref:`zephyr:usb_api`

Reviewer: Remove USB from this one, not used
